packages/@lwc/state/src/index.ts

```diff
@@ -147,6 +147,7 @@ export const defineState: DefineState = <
       private contextConsumptionQueue: Array<
         (runtimeAdapter: ContextRuntimeAdapter<object>) => void
       > = [];
+      private contextUnsubscribes = new Map<symbol, Array<() => void>>();
```

Comment:
This should probably be a `WeakMap`, yes? Otherwise we'll be retaining references to component instances that should be gc'd.